build/dependencies.yaml

```diff
@@ -71,7 +71,7 @@ dependencies:
 
   # Docker
   - name: "docker"
```

Comment:
This entry seems odd / misleading.
- it's only matching one file
- it's not really a dependency?

Reply:
> it's only matching one file

i guess there are not other places to track this.

> it's not really a dependency?

same goes for "crictl" and "cni".

the original idea was to have a machine readable file that can be used by maintainers and users to track what is the latest version of component X that k8s _supports_. k8s may also depend on component X.

based on that, maybe `dependecies.yaml` is not the right name.
but renaming it is a breaking change to the same users.

alternatively all those components that k8s actually does not depend on can be removed.
which is also a breaking change.

Reply:
I guess, we do actually require a minimum docker version for other reasons (building) that we ought to be tracking but that's a separate concern, and this will go away anyhow when dockershim goes away.

Reply:
> and this will go away anyhow when dockershim goes away.

instead i recall discussions to add containerd in this file too, to let users know what is the latest version we are testing.
cc @dims